### PR TITLE
Implement Friends List UI

### LINKED ISSUE #201: Friend List (Event interaction) UI
# Description
Create the UI to view the user friendlist (contact list). (see Figma for more details)
This is part of Event Interaction. 
Ui part for #200 see user story #23 

# Requirements
- [ ] View a list of card of the people whose QrCode the user has scanned 
- [ ] Card should be interactable to view that person profile

### PR DESCRIPTION
**Pull request by jgeaso1266**
## Objective
- implement UI that accompanies VM implemented in #200 

## Key Changes
- implement UI for Friends List
- edit and add UI tests

## Demo

https://github.com/swent-sp-2024-event-radar/event-radar-app/assets/64320145/625c5dc3-67a0-478a-8eed-b43001be90bc

Closes #201

### CODE REVIEW

File: app/src/main/java/com/github/se/eventradar/ui/messages/Messages.kt [hunk 1]
```diff
@@ -350,33 +450,6 @@ fun PreviewMessagesScreen() {
             "https://firebasestorage.googleapis.com/v0/b/event-radar-e6a76.appspot.com/o/Profile_Pictures%2FYJP3bYiaGFPqx64CT6kHOpwvXnv1?alt=media&token=5587f942-efc7-4cbf-920c-7f24a76d7ad1",
             "",
             "johndoe")
-      })
-}
-
-@Preview(showSystemUi = true, showBackground = true)
```
Comment (christinehahaa) [inline app/src/main/java/com/github/se/eventradar/ui/messages/Messages.kt:356]: Was the preview removed for any reason?

**Review by christinehahaa — CHANGES_REQUESTED** [review]
Overall very high quality code. Some questions/comments:
* Does this implementation allow "real time updates" and if so now? i.e. when I scan a friend, it will immediately populate in the friends list. 
* Code duplications are above 3% threshold, ideally fixed before merge
* Is the search bar feature going to be implemented separately?
* Don't forget to merge from main

File: app/src/main/java/com/github/se/eventradar/ui/messages/Messages.kt (hunk 1 above)
Comment (jgeaso1266) [reply app/src/main/java/com/github/se/eventradar/ui/messages/Messages.kt:356]: It was a temporary preview I used that I frankly forgot to remove in my previous PR.

**Review by christinehahaa — CHANGES_REQUESTED** [review]
Thanks for resolving my comments, as discussed offline. Real time updates and search bar look good. Some bugs/nits I found when running on my machine:
* Search bar placeholder says "Search event...". Maybe we can change to a generic "Search" or take in a placeholder string. 
* For some reason, the browse events page doesn't show any events anymore, not sure if that's a related problem?

**Comment by jgeaso1266** [comment]
The duplication is higher than the threshold, but I see no way to refactor this UI component without considerable and complex changes to the code, and duplication remains below 1% once merged into main.

**Review by christinehahaa — APPROVED** [review]
I have tested on machine, and comment regarding duplication noted. LGTM.

**Pull request merged**